src/main/java/ru/yandex/practicum/filmorate/storage/film/FilmDbStorage.java

```diff
@@ -0,0 +1,164 @@
+        KeyHolder keyHolder = new GeneratedKeyHolder();
+        jdbcTemplate.update(connection -> {
+            PreparedStatement stmt = connection.prepareStatement(sqlQuery, new String[]{"film_id"});
+            if (film.getMpa() != null) {
+                stmt.setLong(1, film.getMpa().getId());
+            } else {
+                stmt.setNull(1, Types.BIGINT);
+            }
+            stmt.setString(2, film.getName());
+            stmt.setString(3, film.getDescription());
+            stmt.setDate(4, Date.valueOf(film.getReleaseDate()));
+            stmt.setString(5, String.valueOf(film.getDuration()));
+            return stmt;
+        }, keyHolder);
+        film.setId(keyHolder.getKey().longValue());
+        if (film.getGenres() != null) {
+            String sqlQueryGenres = "insert into FILM_GENRE(FILM_ID, GENRE_ID) values(?, ?)";
+            for (Genre genre : film.getGenres()) {
+                jdbcTemplate.update(sqlQueryGenres, film.getId(), genre.getId());
+            }
+        }
+        return film;
+    }
+
+    @Override
+    public Film updateFilm(Film film) {
+        String sqlQueryFilmUpdate = "update FILM set " +
+                "MPA_ID = ?," +
+                "NAME = ?," +
+                "DESCRIPTION = ?," +
+                "RELEASE_DATE = ?," +
+                "DURATION = ?" +
+                "where FILM_ID = ?";
+        jdbcTemplate.update(sqlQueryFilmUpdate,
+                checkIfMPANotNull(film),
+                film.getName(),
+                film.getDescription(),
+                film.getReleaseDate(),
+                film.getDuration(),
+                film.getId());
+        if (film.getGenres() != null) {
+            String sqlQueryGenresDelete = "delete from FILM_GENRE where FILM_ID = ?";
+            jdbcTemplate.update(sqlQueryGenresDelete, film.getId());
+            String sqlQueryGenresInsert = "insert into FILM_GENRE(FILM_ID, GENRE_ID) values(?, ?)";
+            for (Genre genre : film.getGenres()) {
+                jdbcTemplate.update(sqlQueryGenresInsert, film.getId(), genre.getId());
+            }
+        }
+        return film;
+    }
+
+    private Long checkIfMPANotNull(Film film) {
+        if (film.getMpa() != null) {
+            return film.getMpa().getId();
+        } else return null;
+    }
+
+    @Override
+    public boolean checkIfFilmExists(long filmId) {
+        String sqlQueryFilmSelect = "select count(*) from FILM where FILM_ID = ?";
+        int result = jdbcTemplate.queryForObject(
+                sqlQueryFilmSelect, Integer.class, filmId);
+        return result == 1;
+    }
+
+    @Override
+    public List<Film> getAllFilms() {
+        String sqlQueryFilmsSelect = "select * from FILM as film" +
+                " inner join MPA as mpa" +
+                " ON film.MPA_ID = mpa.MPA_ID";
+        List<Film> listOfFilms = jdbcTemplate.query(sqlQueryFilmsSelect, (resultSet, rowNumber) -> mapRowToFilm(resultSet));
+        return listOfFilms;
```

Comment:
Можно сразу возвращать результат query:
```suggestion
        return jdbcTemplate.query(sqlQueryFilmsSelect, (resultSet, rowNumber) -> mapRowToFilm(resultSet));
```

Reply:
Исправлено.